.github/workflows/ci.yml

```diff
@@ -0,0 +1,57 @@
+name: CI
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+
+      - name: Cache pip
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+          pip install pytest pytest-cov black mypy types-requests pre-commit
+
+      - name: Lint (black --check)
+        run: black --check .
+
+      - name: Type check (mypy)
+        run: mypy --explicit-package-bases src
+
+      - name: Run tests with coverage
+        run: pytest -q --cov=src --cov-report=xml
+
+      - name: Enforce minimum coverage (>=70%)
+        run: |
+          python - <<'PY'
+          import xml.etree.ElementTree as ET, sys
+          r = float(ET.parse('coverage.xml').getroot().get('line-rate'))
+          print(f"Coverage: {r*100:.1f}%")
+          sys.exit(0 if r >= 0.70 else 1)
+          PY
```

Comment:
pytest has `--cov-fail-under` that we should be able to use (e.g. `--cov-fail-under=70`)